### PR TITLE
Dedenpencies update and implementation of smart actions 

### PR DESCRIPTION
**Pull request by carlptr**
- Replaced unmaintained lib request with got
- Updated dependencies with npm audit fix --force
- Added new methods "lock" and "unlock" in Nuki class to use smart actions of Nuki API

### CODE REVIEW

File: lib/bridge.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-var request = require('request-promise');
+//var request = require('request-promise');
```
Comment (Mik13) [inline lib/bridge.js:1]: Could you please remove the line? We won't need it anymore

File: lib/bridge.js [hunk 2]
```diff
@@ -132,7 +130,9 @@ Bridge.prototype.list = function list () {
     ._request('list')
     .then(function (nukis) {
       if (!Array.isArray(nukis)) {
+        console.log('Returned ' + JSON.stringify(nukis));
         throw new Error('did not receive a list of nukis');
+
```
Comment (Mik13) [inline lib/bridge.js:135]: Please move the JSON to the Error-string, so the user can decide if they log the error or not, like:
`throw new Error('Did not receive a list of nukis! Response: ' + JSON.stringify(nukis));`

File: lib/bridge.js [hunk 3]
```diff
@@ -158,6 +158,8 @@ Bridge.prototype.get = function get (nukiId) {
     ._request('list')
     .then(function validateNuki (nukis) {
       if (!Array.isArray(nukis)) {
+        
+        console.log('Returned ' + JSON.stringify(nukis));
         throw new Error('did not receive a list of nukis');
```
Comment (Mik13) [inline lib/bridge.js:163]: Same as last comment

File: lib/bridge.js (hunk 2 above)
Comment (Mik13) [reply lib/bridge.js:135]: And please remove empty lines at the start or end of a block.

File: lib/nuki.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
-var request = require('request-promise');
+//var request = require('request-promise');
```
Comment (Mik13) [inline lib/nuki.js:1]: Remove this line please

**Review by Mik13 — CHANGES_REQUESTED** [review]
Thank you very much, this looks very good. Please change the few things pointed out and I will happily merge it.

**Comment by carlptr** [comment]
Thanks for the helpful review. I made the requested changes.

**Comment by Mik13** [comment]
Thank you very much! I've put your commits into master and published version 1.7.0 to npm.

**Pull request closed without merging**